computer.rb

```diff
@@ -0,0 +1,32 @@
+class Computer
+  def initialize
+    @colors = %w[red blue green orange yellow brown]
+    @last_guess = []
+    @guesses = []
+  end
+
+  def guess(feedback)
+    if !@colors.empty? && feedback.empty?
+      color = @colors.shift
+      guess = [color, color, color, color]
+      @guesses.push(guess)
+      @last_guess = guess
+      guess
+    elsif !@colors.empty? && feedback.length < 4
+      color = @colors.shift
+      guess = @last_guess[0..feedback.length - 1]
+      guess.push(color) while guess.length < 4
+      @guesses.push(guess)
+      @last_guess = guess
+      guess
+    elsif feedback.length == 4
+      loop do
```

Comment:
I do not understand how this is supposed to work, based on the code. Please document what the type of your instance variables are, what `feedback` is, and what we are expecting to see at each stage of the algorithm.

Reply:
@Strzebinczyk You can take a look at https://devhints.io/rdoc for how ruby code is documented. `Rdoc` is the system.

Reply:
Alright, it's better. Were this a production system, I would push for much more thorough explanation of your algorithm.
This is good for now, good job.